### ISSUE TITLE
[BUG] When saving a merge step in Hub Central, the targetEntity is reset to default, which breaks the calculateMergingActivity endpoint

### ISSUE THREAD

**Issue by sgsabbage** [marklogic/marklogic-data-hub#5308]
#### The issue

Short description of the problem:

When saving a merge step in Hub Central, it resets the value of "targetEntity" to "Change this to a valid entity type name; e.g. Customer". As such, in the function [internalCalculateMergingActivity](https://github.com/marklogic/marklogic-data-hub/blob/master/marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mastering/calculateMergingActivityLib.sjs#L33) it will attempt to use that string to identify the target entity for the purpose of providing sources to the Hub Central front end and will return an empty array.

What are the steps to reproduce the issue:
1) Generate a merge step for a known entity with sources stored in the header
2) Edit the value of "targetEntity" in the step JSON
3) Deploy to ML
4) Edit the merge step in Hub Central
5) Pull changes from ML
6) See change to "targetEntity"
7) Edit a merge strategy, priority only contains Length

#### Tech details

Which Operating System are you using?
Windows 10

Which version of MarkLogic are you using?
10

Which version of the Data Hub are you using?
5.4 

Is this a QuickStart UI bug? If so, which browser are you using?
No, Hub Central, Chrome latest.

**Comment by ryanjdew** [comment]
Tracking internally with https://project.marklogic.com/jira/browse/DHFPROD-7002

In the meantime, @sgsabbage a potential workaround would be to rename the targetEntity property to targetEntityType. The latter is the property name we've standardized for specifying the entity.

**Comment by sgsabbage** [comment]
@ryanjdew Many thanks for this. Unfortunately the proposed workaround doesn't work when using Hub Central, as even if you delete the property it will recreate it. As an example, the JSON it's produced currently has both properties in it, e.g.:
```
 {
  "targetEntity": "Change this to a valid entity type name; e.g. Customer",
  "targetEntityType": "http://example.org/Book-0.0.1/Book",
  // Other fields ...
}
```
But the `calculateMergingActivity` endpoint will always take `targetEntity` if it exists. It's not a major blocker as it only stops us from using the source weight dropdown - we can populate it in the JSON if necessary.

**Comment by rjrudin** [comment]
This should be resolved in 5.4.2
